java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToVarHint.java

```diff
@@ -237,6 +239,17 @@ private static boolean isValidVarType(HintContext ctx) {
             return true;
         } else if (expressionTreePath != null) {
             ExecutableElement iterator = ExpandEnhancedForLoop.findIterable(ctx.getInfo());
+            TypeMirror expTypeMirror = ctx.getInfo().getTrees().getTypeMirror(expressionTreePath);
+            TypeMirror typeTypeMirror = ctx.getInfo().getTrees().getTypeMirror(typeTreePath);
+            if (expTypeMirror.getKind() == TypeKind.DECLARED) {
+                DeclaredType dt = (DeclaredType) expTypeMirror;
+                if (dt.getTypeArguments().size() > 0) {
```

Comment:
Please handle array types as well, eg: String[] arr = {"1", "2"}; for (Object o : arr)

Reply:
Also what happens if we iterate over a HashMap<String, Integer> map with expr as map.keySet() or map.values(), we are comparing w first arg type only?

Reply:
Reema, great catch. I have pushed the fix for this. It turns out, that the 2nd case is handed as a declared type itself, because the collection returned by map.keySet() or map.values() would have the type String or Integer as their first type. 
I also have resolved the merge conflicts.